### PR TITLE
Add compatibility for ipywidgets>=8

### PR DESCRIPTION
**Pull request by philippjfr**
- Fixes https://github.com/bokeh/ipywidgets_bokeh/issues/10
- Fixes https://github.com/bokeh/ipywidgets_bokeh/issues/32
- Fixes https://github.com/bokeh/ipywidgets_bokeh/issues/44

### CODE REVIEW

**Comment by philippjfr** [comment]
@mattpap Could you review and release version 1.3 when you get a chance?

We can then follow up with another PR for bokeh 3.0 compatibility.

File: ipywidgets_bokeh/src/manager.ts [hunk 1]
```diff
@@ -1,44 +1,27 @@
...
 let _kernel_id = 0
 
 export class WidgetManager extends HTMLManager {
 
-  private _known_models: {[key: string]: ModelState} = {}
+  private _known_models: {[key: string]: IState} = {}
   private kernel_manager: KernelManager
   private kernel: Kernel.IKernelConnection
   private ws: WebSocket | null = null
+  private _model_objs: { [key: string]: WidgetModel } = {}
```
Comment (philippjfr) [inline ipywidgets_bokeh/src/manager.ts:24]: Might be nice to clean this up when a widget model is destroyed.

**Comment by philippjfr** [comment]
Might need some more testing.

**Comment by mattpap** [comment]
The Python side of this still needs updating. With ipywidgets 8.0.2 I'm getting:
```
export SRC_DIR=/home/mateusz/conda/envs/bk_py310/conda-bld/ipywidgets_bokeh_1669112794075/test_tmp
Traceback (most recent call last):
  File "/home/mateusz/conda/envs/bk_py310/conda-bld/ipywidgets_bokeh_1669112794075/test_tmp/run_test.py", line 2, in <module>
import: 'ipywidgets_bokeh'
    import ipywidgets_bokeh
  File "/home/mateusz/conda/envs/bk_py310/conda-bld/ipywidgets_bokeh_1669112794075/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/lib/python3.11/site-packages/ipywidgets_bokeh/__init__.py", line 1, in <module>
    from .widget import IPyWidget
  File "/home/mateusz/conda/envs/bk_py310/conda-bld/ipywidgets_bokeh_1669112794075/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/lib/python3.11/site-packages/ipywidgets_bokeh/widget.py", line 13, in <module>
    from .kernel import kernel
  File "/home/mateusz/conda/envs/bk_py310/conda-bld/ipywidgets_bokeh_1669112794075/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/lib/python3.11/site-packages/ipywidgets_bokeh/kernel.py", line 142, in <module>
    kernel = BokehKernel.instance()
             ^^^^^^^^^^^^^^^^^^^^^^
  File "/home/mateusz/conda/envs/bk_py310/conda-bld/ipywidgets_bokeh_1669112794075/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/lib/python3.11/site-packages/traitlets/config/configurable.py", line 551, in instance
    inst = cls(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^
  File "/home/mateusz/conda/envs/bk_py310/conda-bld/ipywidgets_bokeh_1669112794075/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/lib/python3.11/site-packages/ipywidgets_bokeh/kernel.py", line 128, in __init__
    self.comm_manager = CommManager(parent=self, kernel=self)
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: CommManager.__init__() got an unexpected keyword argument 'parent'
```
I already published npm and pip ipywidgets_bokeh 1.3.0 before realizing this.

**Comment by mattpap** [comment]
Also, do we expect `ipytwidgets_bokeh` working with both ipywidgets 7.x and 8.x or only 8.x?

**Comment by mattpap** [comment]
Or perhaps this is the issue with ipykernel's `CommManger` in version 6.18. I couldn't reproduce, because I upgraded ipywidgets but not ipykernel, but conda had most recent version in the build environment.

**Comment by mattpap** [comment]
All packages are published now.

**Comment by philippjfr** [comment]
Damn, taking a look now. As far as I could tell it was working with both 7.x and 8.x.

**Comment by mattpap** [comment]
@philippjfr, I removed the defunct version like this `ipykernel 6.*,!=6.18.0` (in `meta.yaml`) and published 1.3.0.

**Comment by philippjfr** [comment]
Perfect, thanks!

**Pull request merged**